### PR TITLE
mongo-3.1 library: add README.md

### PR DESCRIPTION
**Pull request by glebignatieff**
## Motivation

I have spend quite a bit of time reading the sources to understand how to use the library and I would like to share the knowledge with newcomers like myself.

## Solution

Here's a README.md file for the `mongo-3.1` library with the dependencies and basic usage instructions.

I couldn't find any duplicates, so hopefully this could be of use to the project :)

### CODE REVIEW

**Comment by breedx-splk** [comment]
Hi @glebignatieff thanks for the contribution. Can you please run `./gradlew :spotlessApply` to apply the standard formatting and fix up the build failure? Thanks!

File: instrumentation/mongo/mongo-3.1/library/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+Replace OPENTELEMETRY_VERSION with the [latest release](https://central.sonatype.com/search?q=g:io.opentelemetry.instrumentation++a:opentelemetry-mongo-3.1&smo=true).
+
+For Maven, add to your `pom.xml` dependencies:
+
+```xml
+<dependencies>
+  <dependency>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-ktor-2.0</artifactId>
```
Comment (trask) [inline instrumentation/mongo/mongo-3.1/library/README.md:17]:
```suggestion
    <artifactId>opentelemetry-mongo-3.1</artifactId>
```

File: instrumentation/mongo/mongo-3.1/library/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+val mongoTelemetry = MongoTelemetry.builder(openTelemetry).build()
+
+val settings = MongoClientSettings.builder()
+    .applyConnectionString(connectionString)
+    .addCommandListener(mongoTelemetry.newCommandListener())
+    .build()
+
+// With Reactive Streams
+val client = MongoClients.create(settings)
+```
```
Comment (trask) [inline instrumentation/mongo/mongo-3.1/library/README.md:44]: can you change the example from Kotlin to Java? thx

File: instrumentation/mongo/mongo-3.1/library/README.md (hunk 1 above)
Comment (glebignatieff) [reply instrumentation/mongo/mongo-3.1/library/README.md:17]: Whoops, my bad.

File: instrumentation/mongo/mongo-3.1/library/README.md (hunk 2 above)
Comment (glebignatieff) [reply instrumentation/mongo/mongo-3.1/library/README.md]: [Fixed](https://github.com/open-telemetry/opentelemetry-java-instrumentation/compare/ba398725736b220bfd5854f95b899deef6705497..37cd7121a20773e604b15429c1ec733834455835).

**Pull request merged**